lib/Doctrine/DBAL/DriverManager.php

```diff
@@ -115,11 +115,16 @@ private function __construct()
      * <b>driverClass</b>:
      * The driver class to use.
      *
-     * @param mixed[]            $params       The parameters.
-     * @param Configuration|null $config       The configuration to use.
-     * @param EventManager|null  $eventManager The event manager to use.
+     * @param array{wrapperClass?: class-string<T>} $params
+     * @param Configuration|null                    $config       The configuration to use.
+     * @param EventManager|null                     $eventManager The event manager to use.
      *
      * @throws DBALException
+     *
+     * @phpstan-param mixed[] $params
+     * @phpstan-return Connection
```

Comment:
This is getting ugly. You need to tell each SA what's happening in its own language. I hope it will force us to introduce simpler and cleaner APIs.

Reply:
Why is `@phpstan-return` needed? There's no `@return` and there is a return type declaration with the same type. Does PHPStan get confused with `@psalm-return` and needs an override?

Reply:
Removing the first one causes this:

```
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   Doctrine/DBAL/Id/TableGenerator.php                                                                                                                                               
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- 
  75     Parameter #1 $params of static method Doctrine\DBAL\DriverManager::getConnection() expects array()|array('wrapperClass' => class-string<Doctrine\DBAL\Connection>), array given.  
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- 

 ------ -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   Doctrine/DBAL/Schema/SqliteSchemaManager.php                                                                                                                                                            
 ------ -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- 
  59     Parameter #1 $params of static method Doctrine\DBAL\DriverManager::getConnection() expects array()|array('wrapperClass' => class-string<Doctrine\DBAL\Connection>), array('driver' => mixed, 'path' =>  
         string) given.                                                                                                                                                                                          
 ------ -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- 
```
I will remove `@phpstan-return`, it is not needed, it's a remnant of when I was trying to get phpstan to be quiet about that warning I added in the ignore list.